### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_tabs-js/)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -9,32 +11,52 @@ export const tabs = [
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
-export const App = () => (
-  <div className="section">
-    <h1 className="title">
-      Selected tab is Tab 1
-    </h1>
+export const App = () => {
+  const [value, setValue] = useState('tab-1');
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:15]:
```suggestion
  const [selectedTabId, set
  selectedTabId] = useState(tabs[0].id);
```
better to set id from first tabs element
consider more meaningful naming

File: src/App.jsx [hunk 2]
```diff
@@ -9,32 +11,52 @@ export const tabs = [
...
+  const [titleH1, setTitleH1] = useState('Tab 1');
+  const [text, setText] = useState('Some text 1');
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:17]:
```suggestion
```
redundant states.
just manage selectedTabId and find selectedTab by id.
```
const selectedTab = tabs.find(tab => tab.id === selectedTabId);
```
another way keep in state full tab instead only tab id.

File: src/App.jsx [hunk 3]
```diff
@@ -9,32 +11,52 @@ export const tabs = [
...
+                  })}
+                >
+                  <a
+                    href={`#${id}`}
+                    data-cy="TabLink"
+                    onClick={() => {
+                      setValue(id);
+                      setTitleH1(title);
+                      setText(content);
+                    }}
```
Comment (VitaliyBondarenko1982) [inline src/App.jsx:46]:
```
const selectTab = () => setSelectedTabId(id);
```

```suggestion
                    onClick={selectTab}
```

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Needs some improvements.

File: src/components/Tabs/Tabs.jsx [hunk 4]
```diff
@@ -1 +1,39 @@
...
+                className={classnames({
+                  'is-active': selectedTab.id === tab.id,
+                })}
+              >
+                <a
+                  href={`#${tab.id}`}
+                  data-cy="TabLink"
+                  onClick={() => selectedTab.id !== tab.id
+                    && onTabSelected(tab)
+                  }
```
Comment (loralevitska) [inline src/components/Tabs/Tabs.jsx:25]: Move to separate function  (transfer the function to a variable, and here only write the name of the function)

File: src/components/Tabs/Tabs.jsx [hunk 5]
```diff
@@ -1 +1,39 @@
...
+            tabs.map(tab => (
+              <li
+                key={tab.id}
+                data-cy="Tab"
+                className={classnames({
+                  'is-active': selectedTab.id === tab.id,
+                })}
+              >
+                <a
+                  href={`#${tab.id}`}
```
Comment (loralevitska) [inline src/components/Tabs/Tabs.jsx:21]: Destructure `tab`

**Review by loralevitska — APPROVED** [review]
Well done) Check my comments for improvement

**Pull request closed without merging**